tck/src/main/java/org/eclipse/microprofile/rest/client/tck/ClientReuseTest.java

```diff
@@ -0,0 +1,121 @@
+/*
+ * Copyright (c) 2017 Contributors to the Eclipse Foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.eclipse.microprofile.rest.client.tck;
+
+import org.eclipse.microprofile.rest.client.RestClientBuilder;
+import org.eclipse.microprofile.rest.client.tck.interfaces.JsonPClient;
+import org.jboss.arquillian.container.test.api.Deployment;
+import org.jboss.shrinkwrap.api.Archive;
+import org.jboss.shrinkwrap.api.ShrinkWrap;
+import org.jboss.shrinkwrap.api.spec.WebArchive;
+import org.junit.Assert;
+import org.testng.annotations.Test;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.concurrent.Callable;
+import java.util.concurrent.CancellationException;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.Future;
+import java.util.concurrent.TimeUnit;
+
+import static com.github.tomakehurst.wiremock.client.WireMock.aResponse;
+import static com.github.tomakehurst.wiremock.client.WireMock.get;
+import static com.github.tomakehurst.wiremock.client.WireMock.stubFor;
+import static com.github.tomakehurst.wiremock.client.WireMock.urlEqualTo;
+import static org.testng.Assert.assertTrue;
+import static org.testng.Assert.fail;
+
+/**
+ * @author Michal Szynkiewicz, michal.l.szynkiewicz@gmail.com
+ * <br>
+ * Date: 11/30/18
+ */
+public class ClientReuseTest extends WiremockArquillianTest {
+
+    @Deployment
+    public static Archive<?> createDeployment() {
+        return ShrinkWrap.create(WebArchive.class, ClientReuseTest.class.getSimpleName() + ".war")
+            .addClasses(WiremockArquillianTest.class, JsonPClient.class);
+    }
+
+    @Test
+    public void shouldReuseClientAfterFailure() throws Throwable {
+       callWithTimeout(this::testReuseClientAfterFailure, 20, TimeUnit.SECONDS);
+    }
+
+    private Void testReuseClientAfterFailure() {
+        JsonPClient client =
+            RestClientBuilder.newBuilder()
+                .baseUri(getServerURI())
+                .build(JsonPClient.class);
+
+        performSuccessfulRequest(client);
+
+        performFailingRequest(client);
+
+        performSuccessfulRequest(client);
+        return null;
+    }
+
+    private void performSuccessfulRequest(JsonPClient client) {
+        stubReturning("{\"content\": true}");
```

Comment:
For both of the calls to `stubReturning` (here and in the `performFailingRequest` method), I ran into problems in my local environment.  I think WireMock must use thread locals that are not copied over to the thread in the ExecutorService.  I was able to make this work by calling `stubReturning` from the main test method (`shouldReuseClientAfterFailure`) before calling the `callWithTimeout` method.  This also required that I use different URIs (IDs) so that I could set them all in advance.  Here is a short summary of my changes:

```
    @Test
    public void shouldReuseClientAfterFailure() throws Throwable {
        stubReturning(1, "{\"content\": true}");
        stubReturning(2, "Not a json");
        stubReturning(3, "{\"content2\": true}");
        callWithTimeout(this::testReuseClientAfterFailure, 20, TimeUnit.SECONDS);
    }

    private Void testReuseClientAfterFailure() {
        JsonPClient client =
            RestClientBuilder.newBuilder()
                .baseUri(getServerURI())
                .build(JsonPClient.class);

        //expect successful invocation
        assertTrue(client.get("1").getBoolean("content"));

        //expect failing invocation
        expectFailure(() -> client.get("2"));

        //expect successful invocation
        assertTrue(client.get("3").getBoolean("content2"));
        return null;
    }

    private void stubReturning(int id, String text) {
        stubFor(
            get(urlEqualTo("/" + id))
                .willReturn(
                    aResponse().withBody(text).withHeader("Content-Type", "application/json")
                )
        );
    }
...
```

With these changes, this test passes for me in CXF.

Reply:
@andymc12 thanks. It works as it is for SmallRye & RestEasy but we're using slightly newer versions of testng, etc, so this may be to blame. 
I will update it.

Reply:
That could be, or possibly SmallRye/RestEasy are testing with wiremock on the default port (8080).  I noticed in the CXF testing environment, that the wiremock client was attempting to connect on port 8080 (I'm guess that that is the default port), but pom file specified port 8765, so I was getting a connect exception.

Reply:
done